src/langcheck/metrics/zh/_tokenizers.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import annotations
+
+import abc
+from collections.abc import Iterator
+
+from rouge_score.tokenizers import Tokenizer as BaseTokenizer
+
+try:
+    import hanlp
+
+    # size 43M+, fine grained tokenizer
+    DEFAULT_TOKENIZER_WEIGHT = hanlp.pretrained.tok.FINE_ELECTRA_SMALL_ZH  # type: ignore[reportGeneralTypeIssues] # noqa: E501
+except ModuleNotFoundError:
+    raise ModuleNotFoundError("No module named 'HanLP'.\n"
+                              "Install it by pip install -U hanlp")
+
+# Chinese
+# https://github.com/yikeke/zh-style-guide/blob/master/source/%E6%A0%87%E7%82%B9%E7%AC%A6%E5%8F%B7/%E5%B8%B8%E7%94%A8%E4%B8%AD%E6%96%87%E6%A0%87%E7%82%B9%E7%AC%A6%E5%8F%B7.md
+_PUNCTUATIONS = [
+    '、', '，', '。', '：', '；', '?', '!', "？", "！", '～', '-', '—', '——', '……',
+    '⋯⋯', '/'
+]
+
+
+class _ChineseTokenizer(BaseTokenizer):
+
+    @abc.abstractmethod
+    def _tokenize(self, text: str) -> Iterator[str]:
+        raise NotImplementedError(
+            "Tokenizer for Chinese must override `_tokenize()` method")
+
+    def tokenize(self, text: str) -> list[str]:
+        tokens = self._tokenize(text)
+        return [
+            token for token in tokens if (token and token not in _PUNCTUATIONS)
+        ]
+
+
+class HanLPTokenizer(_ChineseTokenizer):
+    '''HanLP based Tokenizer for Chinese.
+
+    The default tokenizer to calculate rouge score base on HanLP.
+
+    .. note::
+        HanLP https://github.com/hankcs/HanLP/tree/doc-zh
+        HanLP is the newest tokenizer that have developer maintaince.
+        1.HanLP have multi-task mode and single task mode. Multitask
+        Model need download 400MB+  pretrained weight, in contrast,
+        single task only need 40MB+. Use single task mode in default.
+        2.LLM generated content have lot of sentences in most situtation.
+        use HanLP pipeline mode for concurrency.
```

Comment:
1. I downloaded one of them, and I think it makes langcheck become over weight if use multi-task model[hanlp multi-task model pretrained weight list](https://hanlp.hankcs.com/docs/api/hanlp/pretrained/mtl.html)
2. hanlp lib pretrained model for tokenization.
[hanlp single task pretrained weight list](https://hanlp.hankcs.com/docs/api/hanlp/pretrained/tok.html )
3. mentioned in a notebook provided by in chapter:
> 并行分词
[concurrency hanlp](https://github.com/hankcs/HanLP/blob/doc-zh/plugins/hanlp_demo/hanlp_demo/zh/tok_stl.ipynb)

Reply:
Thanks for the additional context! Based on that, here's my suggestion on rewording this part a bit - please take a look!

```suggestion
        We run HanLP's single-task models using HanLP's pipeline mode, because:
        1. HanLP has both multi-task models and single-task models. The
        multi-task models are quite large (generally 400MB+), whereas the
        single-task models are only ~40MB. So, we use a single-task model by
        default.
        2. HanLP's pipeline mode allows processing of long texts (i.e. many
        sentences) efficiently in parallel. It splits long text into sentences
        and applies the tokenizer to the sentences in parallel.
```